diff --git a/src/Baballonia.Desktop/Baballonia.Desktop.csproj b/src/Baballonia.Desktop/Baballonia.Desktop.csproj
index 3f351ec..6e77181 100644
--- a/src/Baballonia.Desktop/Baballonia.Desktop.csproj
+++ b/src/Baballonia.Desktop/Baballonia.Desktop.csproj
@@ -48,7 +48,9 @@
 
         <PackageReference Include="Project-Babble.OpenCvSharp4.mini.runtime.ubuntu.22.04-arm64" Version="4.11.0.1" Condition="$([MSBuild]::IsOSPlatform('Linux')) And
              $([System.Runtime.InteropServices.RuntimeInformation]::OSArchitecture.ToString().Contains('Arm'))" />
-	    <PackageReference Include="Microsoft.ML.OnnxRuntime" Version="1.22.0" Condition="$([MSBuild]::IsOSPlatform('Linux'))" />
+        <PackageReference Include="Microsoft.ML.OnnxRuntime.Managed" Version="1.26.0" Condition="$([MSBuild]::IsOSPlatform('Linux'))">
+            <ExcludeAssets>native</ExcludeAssets>
+        </PackageReference>
         <PackageReference Include="SkiaSharp.NativeAssets.Linux" Version="3.119.0" Condition="$([MSBuild]::IsOSPlatform('Linux'))" />
         <PackageReference Include="Xaml.Behaviors.Interactions.Draggable" Version="11.3.9.5" />
         <ProjectReference Include="..\Baballonia.LibV4L2Capture\Baballonia.LibV4L2Capture.csproj" Condition="$([MSBuild]::IsOSPlatform('Linux'))" />
diff --git a/src/Baballonia/Baballonia.csproj b/src/Baballonia/Baballonia.csproj
index 453c4bf..ca941b3 100644
--- a/src/Baballonia/Baballonia.csproj
+++ b/src/Baballonia/Baballonia.csproj
@@ -64,6 +64,9 @@
         <PackageReference Include="Microsoft.Extensions.Logging.Abstractions" Version="10.0.4" />
         <PackageReference Include="Microsoft.ML" Version="4.0.3" />
         <PackageReference Include="Microsoft.ML.OnnxTransformer" Version="4.0.3" />
+        <PackageReference Include="Microsoft.ML.OnnxRuntime.Managed" Version="1.26.0">
+            <ExcludeAssets>native</ExcludeAssets>
+        </PackageReference>
         <PackageReference Include="Newtonsoft.Json" Version="13.0.4" />
         <PackageReference Include="OpenCvSharp4" Version="4.13.0.20260308" />
         <PackageReference Include="OscCore" Version="1.0.5" />
